public/pages/DetectorResults/containers/AnomalyHistory.tsx

```diff
@@ -173,6 +173,12 @@ export const AnomalyHistory = (props: AnomalyHistoryProps) => {
   const backgroundColor = darkModeEnabled() ? '#29017' : '#F7F7F7';
   const resultIndex = get(props, 'detector.resultIndex', '');
 
+  // Utility fn to only fetch data when either it is non-historical, or historical and
+  // there is a populated task ID which will be used to fetch the historical results
+  const isValidToFetch = () => {
```

Comment:
if it is not hisotorical even on first load there will be a detector ID?

Reply:
Yes- the `AnomalyHistory` component is conditionally rendered based on a valid detector, so there will always be a detector ID. (additionally it is required in the url path which cannot be empty). The edge case is that the task ID is asynchronously fetched after the detector ID (via get detector details API) to populate it, which is why the initial render is happening without a task ID.

A different option is to refactor some of the `HistoricalDetectorResults` page to try to prevent rendering the entire `AnomalyHistory` component until the task ID is valid. But the current way may provide a better user experience in that it is more react-like, where the only re-rendering is happening in the charts component where the task ID is used to actually populate it once it is fetched and data is retrieved.